### PR TITLE
build: shrink the demo docker image size

### LINKED ISSUE #329: Reduce the size of docker image
### Problem

The size of docker image of OpenMLDB currently provided on docker hub is too large compare to MySQL or PostgresQL (100MB+ vs 2.2GB). It takes nearly 1 and half hour to pulling it from docker hub.

After digging into the image(the output of ```du``` below), i found that the ```spark-3.0.0-bin-openmldb``` directory takes nearly 77% of the space and a Python script named  ```openmldb_batch.py```  depends on it.
```
[root@7f8f08523050 work]# du -h -d 1
2.3M	./taxi-trip
1.7G	./openmldb
62M	./zookeeper-3.4.14
1.8G	.
[root@7f8f08523050 work]# cd openmldb/
[root@7f8f08523050 openmldb]# dh -h -d 1
bash: dh: command not found
[root@7f8f08523050 openmldb]# cd ..
[root@7f8f08523050 work]# ls
develop_ml_application_tour.ipynb  openmldb  taxi-trip	zookeeper-3.4.14
[root@7f8f08523050 work]# du -h -d 1
2.3M	./taxi-trip
1.7G	./openmldb
62M	./zookeeper-3.4.14
1.8G	.
[root@7f8f08523050 work]# cd openmldb/
[root@7f8f08523050 openmldb]# du -h -d 1
72K	./tools
16K	./conf
1.2G	./batch
488M	./bin
1.7G	.
[root@7f8f08523050 openmldb]# cd batch/spark-3.0.0-bin-openmldb/
[root@7f8f08523050 spark-3.0.0-bin-openmldb]# du -h -d 1
352K	./licenses
4.0M	./examples
1.1M	./data
10M	./yarn
40K	./conf
802M	./python
120K	./bin
405M	./jars
88K	./sbin
1.2G	.
```
However, there should be an image for those who just want to play with OpenMLDB itself rather than the **taxi-trip** demo.

Another reason for a smaller image is that develops may use OpenMLDB image as a base image to build their own app image, so there should be an image to meet this requirement.

### Solution

My suggestion is providing two images named ```openmldb``` and ```openmldb-demo```.

The ```openmldb``` image should contains minimal requirement to bring up a OpenMLDB instance. With binary, script and SDK inside the image, i hope the size the image would be no more than 600MB.
The ```openmldb-demo``` should contains the content of taxi-trip demo as the current ```openmldb``` on docker hub.

### PR DESCRIPTION
**Pull request by zhanghaohit**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
demo docker image

* **What is the current behavior?** (You can also link to an open issue here)
current docker image is too large #329

* **What is the new behavior (if this is a feature change)?**
docker image shrink from ~3GB to ~1.3 GB

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: demo/predict-taxi-trip-duration-nb/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,137 @@
-FROM centos:8
+# https://github.com/docker-library/openjdk/blob/61bfbb30ae5a0395f08d41107b86bcce90d70ba6/11/jre/slim-bullseye/Dockerfile
+FROM openjdk:11.0.13-jre-slim-bullseye
 
-LABEL org.opencontainers.image.source https://github.com/4paradigm/OpenMLDB
```
Comment (aceforeverd) [inline demo/predict-taxi-trip-duration-nb/Dockerfile:3]: keep this line

File: demo/predict-taxi-trip-duration-nb/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,137 @@
...
 
-RUN yum install -y python38 python38-pip python38-numpy libgomp java-1.8.0-openjdk-headless && \
-    yum clean all && \
-    pip3 install lightgbm tornado requests pandas openmldb
+RUN set -ex \
+	\
+	&& savedAptMark="$(apt-mark showmanual)" \
+	&& apt-get update && apt-get install -y --no-install-recommends \
+		dpkg-dev \
+		gcc \
```
Comment (aceforeverd) [inline demo/predict-taxi-trip-duration-nb/Dockerfile]: python compile dependency may not included in final image
can we use multi-stage build ?

File: demo/predict-taxi-trip-duration-nb/Dockerfile [hunk 3]
```diff
@@ -1,10 +1,137 @@
...
+		\( \
+			\( -type d -a \( -name test -o -name tests -o -name idle_test \) \) \
+			-o \
+			\( -type f -a \( -name '*.pyc' -o -name '*.pyo' \) \) \
+		\) -exec rm -rf '{}' +; \
+	rm -f get-pip.py
+
+
+RUN  apt-get update \
+     && apt-get install -y --no-install-recommends libgomp1 curl binutils vim procps \
```
Comment (aceforeverd) [inline demo/predict-taxi-trip-duration-nb/Dockerfile]: vim redunant

File: demo/predict-taxi-trip-duration-nb/Dockerfile [hunk 4]
```diff
@@ -1,10 +1,137 @@
-FROM centos:8
+# https://github.com/docker-library/openjdk/blob/61bfbb30ae5a0395f08d41107b86bcce90d70ba6/11/jre/slim-bullseye/Dockerfile
+FROM openjdk:11.0.13-jre-slim-bullseye
```
Comment (aceforeverd) [inline demo/predict-taxi-trip-duration-nb/Dockerfile:2]: I'm open for Debian but is there any reason to switch ?
And code compile on centos ...

**Review by aceforeverd — COMMENTED** [review]
keep a eye on https://github.com/4paradigm/OpenMLDB/blob/569e9883887542036af9f70e2a6e6e6a1f50aabb/demo/predict-taxi-trip-duration-nb/docker-compose.test.yml
this is the test file

File: demo/predict-taxi-trip-duration-nb/Dockerfile (hunk 4 above)
Comment (zhanghaohit) [reply demo/predict-taxi-trip-duration-nb/Dockerfile:2]: debian base seems smaller than centos

Comment (zhanghaohit) [reply demo/predict-taxi-trip-duration-nb/Dockerfile:2]: Total image size:
centos:8: 1.63G
openjdk:11.0.13-jre-slim-bullseye: 1.35GB

File: demo/predict-taxi-trip-duration-nb/Dockerfile (hunk 2 above)
Comment (zhanghaohit) [reply demo/predict-taxi-trip-duration-nb/Dockerfile]: The python compile dependency deps will be auto purged by apt-mark.

**Review by dl239 — APPROVED** [review]

**Comment by aceforeverd** [comment]
@dl239 
- `apt-get install python3` if it works?
- is that ok to run over jdk11 ?

**Review by aceforeverd — APPROVED** [review]

File: demo/predict-taxi-trip-duration-nb/Dockerfile (hunk 4 above)
Comment (aceforeverd) [reply demo/predict-taxi-trip-duration-nb/Dockerfile:2]: looks like slim did it work

**Pull request merged**